src/translator/sentence_ranges.cpp

```diff
@@ -99,5 +99,42 @@ string_view AnnotatedText::asStringView(const ByteRange &byteRange) const {
   return string_view(data, size);
 }
 
+string_view AnnotatedText::pre(size_t sentenceIdx) const {
+  // Find start of filler-text before, there's a corner case when there's no
+  // sentence before.
+  const char *start{nullptr};
+  if (sentenceIdx == 0) {
+    // If first sentence, filler begins at start of whole-text.
+    start = &(text[0]);
```

Comment:
unnecessary brackets here

Reply:
I'm worried this might lead to segfaults. [0], is there a safe way to avoid the indexing and get a char*?

Reply:
`const char * start = nullptr`

Reply:
Oh sorry wrong comment. The [] takes precedence before &: https://en.cppreference.com/w/cpp/language/operator_precedence

Reply:
text.data()

Reply:
x.data() is used inside the container now.